models/Movie.model.js

```diff
@@ -0,0 +1,47 @@
+const { Schema, model } = require("mongoose");
+
+const movieSchema = new Schema(
+  {
+    title: {
+      type: String,
+      require: true
+    },
+    plot: {
+      type: String,
+    },
+    genre: {
+      type: String,
+    },
+    year: {
+      type: String,
+    },
+    director: {
+      type: String,
+    },
+    cast: {
+      type: String,
+    },
+    trailer: {
+      type: String,
+    },
+    rating: {
+      type: String,
+    },
```

Comment:
That should be a number, probably a decimal.

```suggestion
rating: Schema.Types.Decimal128
```

see also https://mongoosejs.com/docs/schematypes.html